### PR TITLE
Add record info for postfailhooks and check proper variable

### PR DESCRIPTION
**Pull request by foursixnine**
While having the post fail hooks is very useful, specially when they are called in chain, it' s no fun to try to figure out what happened or when the actual post fail hook started, let' s use a record info for that? :)

On the other hand, sneak in a proper check for ARCH instead of MACHINE in the user_defined_snapshot test

### CODE REVIEW

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -564,6 +564,7 @@ sub remount_tmp_if_ro {
 ## of wait_boot, e.g. see `git grep -l reboot | xargs grep -L wait_boot`
 sub post_fail_hook {
     my ($self) = @_;
+    record_info('post fail hook', 'Post fail hook started for test distribution.');
```
Comment (okurz) [inline lib/opensusebasetest.pm]: Don't we have a box just in before stating "Failed"?

File: tests/x11/user_defined_snapshot.pm [hunk 2]
```diff
@@ -55,7 +55,7 @@ sub run {
     send_key "alt-l";
     $self->{in_wait_boot} = 1;
     power_action('reboot', keepconsole => 1, textmode => 1);
-    $self->handle_uefi_boot_disk_workaround() if get_var('MACHINE') =~ qr'aarch64';
+    $self->handle_uefi_boot_disk_workaround() if check_var('ARCH', 'aarch64');
```
Comment (okurz) [inline tests/x11/user_defined_snapshot.pm]: Hm, could you please check the history if we are changing back and forth here?

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (foursixnine) [reply lib/opensusebasetest.pm]: Yes, however iirc doesn't happen everywhere, I will just double check before pushing more :)

File: tests/x11/user_defined_snapshot.pm (hunk 2 above)
Comment (foursixnine) [reply tests/x11/user_defined_snapshot.pm]: Should be check_var, but for the time being, let's not invest more time on this.

**Pull request closed without merging**